taca/deliver/deliver.py

```diff
@@ -0,0 +1,385 @@
+            according to the config.
+            
+            The config should contain the key 'files_to_deliver', which should
+            be a list of tuples with source path patterns and destination path
+            patterns. The source path can be a file glob and can refer to a 
+            folder or file. File globs will be expanded and folders will be
+            traversed to include everything beneath.
+             
+            :returns: A generator of tuples with source path, 
+                destination path and the checksum of the source file 
+                (or None if source is a folder)
+        """
+        for sfile, dfile in getattr(self,'files_to_deliver',[]):
+            for f in glob.iglob(self.expand_path(sfile)):
+                if (os.path.isdir(f)):
+                    # walk over all folders and files below
+                    for pdir,_,files in os.walk(f):
+                        for current in files:
+                            fpath = os.path.join(pdir,current)
+                            # use the relative path for the destination path
+                            fname = os.path.relpath(fpath,f)
+                            yield(fpath,
+                                os.path.join(self.expand_path(dfile),fname),
+                                hashfile(fpath,hasher=self.hash_algorithm) \
+                                if not self.no_checksum else None)
+                else:
+                    yield (f, 
+                        os.path.join(self.expand_path(dfile),os.path.basename(f)), 
+                        hashfile(f,hasher=self.hash_algorithm) \
+                        if not self.no_checksum else None)
+    
+    def stage_delivery(self):
+        """ Stage a delivery by symlinking source paths to destination paths 
+            according to the returned tuples from the gather_files function. 
+            Checksums will be written to a digest file in the staging path. 
+            Failure to stage individual files will be logged as warnings but will
+            not terminate the staging. 
+            
+            :raises DelivererError: if an unexpected error occurred
+        """
+        digestpath = self.staging_digestfile()
+        create_folder(os.path.dirname(digestpath))
+        try: 
+            with open(digestpath,'w') as dh:
+                agent = transfer.SymlinkAgent(None,None,relative=True,log=self.log)
+                for src, dst, digest in self.gather_files():
+                    agent.src_path = src
+                    agent.dest_path = dst
+                    try:
+                        agent.transfer()
+                    except transfer.TransferError as e:
+                        self.log.warning("failed to stage file '{}' when "\
+                            "delivering {} - reason: {}".format(
+                                src,str(self),e))
+                    except transfer.SymlinkError as e:
```

Comment:
you can catch different exceptions in one single line if you're going to treat them as equal:

```python
try:
    agent.transfer()
except (transfer.TransferError, transfer.SymlinkError) as e:
    self.log.warning("failed to stage file '{}' when "\
        "delivering {} - reason: {}".format(
            src,str(self),e))
```